// @backend-ai/sdk-express/dist/core-client.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoreClient = void 0;
const grpc = __importStar(require("@grpc/grpc-js"));
// Load proto files from SDK package directory
const path = require('path');
const protoDir = path.join(__dirname, '../protos');
const toolGrpc = require(path.join(protoDir, 'tool_grpc_pb'));
const toolProto = require(path.join(protoDir, 'tool_pb'));
const Struct = require('google-protobuf/google/protobuf/struct_pb').Struct;
/**
 * Core Service Client
 * Handles communication with the Backend AI Core service
 */
class CoreClient {
    constructor(verbose = false) {
        this.verbose = verbose;
    }
    /**
     * Connect to Core service
     */
    connect(coreUrl) {
        try {
            this.client = new toolGrpc.ToolRegistryServiceClient(coreUrl, grpc.credentials.createInsecure());
            if (this.verbose) {
                console.log(`✅ Connected to Core at ${coreUrl}`);
            }
        }
        catch (error) {
            console.error('❌ Failed to connect to Core:', error);
            throw error;
        }
    }
    /**
     * Register tools with Core
     */
    async registerTools(tenantId, adapterId, tools) {
        return new Promise((resolve, reject) => {
            try {
                const req = new toolProto.RegisterToolsRequest();
                req.setTenantId(tenantId);
                req.setAdapterId(adapterId);
                const toolDefs = tools.map((t) => {
                    const def = new toolProto.ToolDefinition();
                    def.setToolId(t.tool_id);
                    def.setName(t.name);
                    def.setDescription(t.description);
                    // Convert schemas to Struct
                    const inputStruct = Struct.fromJavaScript(t.input_schema || {});
                    const outputStruct = Struct.fromJavaScript(t.response_schema || {});
                    def.setInputSchema(inputStruct);
                    def.setOutputSchema(outputStruct);
                    return def;
                });
                req.setToolsList(toolDefs);
                this.client.registerTools(req, (err, response) => {
                    if (err) {
                        console.error('❌ Tool registration failed:', err);
                        return reject(err);
                    }
                    if (this.verbose) {
                        console.log(`✅ Successfully registered ${tools.length} tools with Core`);
                    }
                    resolve();
                });
            }
            catch (error) {
                console.error('❌ Error during tool registration:', error);
                reject(error);
            }
        });
    }
    /**
     * Verify connection to Core
     */
    async ping() {
        try {
            // Implement ping if Core has health check endpoint
            return true;
        }
        catch {
            return false;
        }
    }
}
exports.CoreClient = CoreClient;
