// @backend-ai/sdk-express/dist/executor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressExecutor = void 0;
/**
 * Express Executor
 * Executes Express route handlers with mock req/res objects
 */
class ExpressExecutor {
    constructor(routerMap) {
        this.routerMap = routerMap;
    }
    /**
     * Execute a tool by ID
     */
    async execute(toolId, input) {
        const mapping = this.routerMap.get(toolId);
        if (!mapping) {
            throw new Error(`Tool not found: ${toolId}`);
        }
        const { handler, path, method } = mapping;
        // Parse path parameters
        const paramKeys = this.extractParamKeys(path);
        // Distribute input to req parts
        const { params, query, body } = this.distributeInput(input, paramKeys, method);
        // Create mock Express request
        const req = {
            body,
            params,
            query,
            headers: {},
            method,
            path,
        };
        // Create mock Express response
        return new Promise((resolve, reject) => {
            let status = 200;
            const res = {
                status(code) {
                    status = code;
                    return this;
                },
                json: (data) => {
                    const safeData = data && typeof data === 'object'
                        ? JSON.parse(JSON.stringify(data))
                        : {};
                    resolve({
                        status,
                        body: safeData,
                    });
                },
                send: (data) => {
                    const safeData = data && typeof data === 'object'
                        ? JSON.parse(JSON.stringify(data))
                        : {};
                    resolve({
                        status,
                        body: safeData,
                    });
                },
            };
            try {
                const result = handler(req, res);
                // Handle async handlers
                if (result instanceof Promise) {
                    result.catch(reject);
                }
            }
            catch (err) {
                reject(err);
            }
        });
    }
    /**
     * Extract parameter keys from path
     */
    extractParamKeys(path) {
        const keys = [];
        const parts = path.split('/');
        for (const part of parts) {
            if (part.startsWith(':')) {
                const key = part.substring(1).replace('?', '');
                keys.push(key);
            }
        }
        return keys;
    }
    /**
     * Distribute input to req.params, req.query, req.body
     */
    distributeInput(input, paramKeys, method) {
        const params = {};
        const query = {};
        const body = {};
        const rawInput = input?.body || input || {};
        for (const [key, value] of Object.entries(rawInput)) {
            if (paramKeys.includes(key)) {
                params[key] = value;
            }
            else if (method === 'GET' || method === 'DELETE') {
                query[key] = value;
            }
            else {
                body[key] = value;
            }
        }
        return { params, query, body };
    }
}
exports.ExpressExecutor = ExpressExecutor;
