// @backend-ai/sdk-express/dist/executor-server.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExecutorServer = void 0;
const grpc = __importStar(require("@grpc/grpc-js"));
const executor_1 = require("./executor");
// Load proto files from SDK package directory
const path = require('path');
const protoDir = path.join(__dirname, '../protos');
const executeGrpc = require(path.join(protoDir, 'execute_grpc_pb'));
const executeProto = require(path.join(protoDir, 'execute_pb'));
const Struct = require('google-protobuf/google/protobuf/struct_pb').Struct;
/**
 * gRPC Executor Server
 * Receives tool execution requests from Core and executes Express handlers
 */
class ExecutorServer {
    constructor(routerMap, port = 6000, verbose = false) {
        this.port = port;
        this.server = new grpc.Server();
        this.executor = new executor_1.ExpressExecutor(routerMap);
        this.verbose = verbose;
    }
    /**
     * Start the gRPC server
     */
    start() {
        // Register service
        this.server.addService(executeGrpc.ToolExecutionServiceService, {
            executeTool: this.handleExecuteTool.bind(this),
        });
        // Bind to port
        this.server.bindAsync(`0.0.0.0:${this.port}`, grpc.ServerCredentials.createInsecure(), (err, port) => {
            if (err) {
                console.error('❌ Failed to start executor server:', err);
                throw err;
            }
            console.log(`🧩 SDK Executor running on port ${port}`);
        });
    }
    /**
     * Handle tool execution request
     */
    async handleExecuteTool(call, callback) {
        try {
            const toolId = call.request.getToolId();
            const inputStruct = call.request.getInput();
            const input = inputStruct ? inputStruct.toJavaScript() : {};
            if (this.verbose) {
                console.log(`[Executor] Executing tool: ${toolId}`);
                console.log(`[Executor] Input:`, input);
            }
            // Execute the tool
            const result = await this.executor.execute(toolId, input);
            if (this.verbose) {
                console.log(`[Executor] Result:`, result);
            }
            // Create response
            const outputStruct = Struct.fromJavaScript(result || {});
            const response = new executeProto.ToolExecutionResponse();
            response.setSuccess(true);
            response.setOutput(outputStruct);
            callback(null, response);
        }
        catch (err) {
            console.error('❌ Tool execution failed:', err);
            const response = new executeProto.ToolExecutionResponse();
            response.setSuccess(false);
            response.setErrorMessage(err?.message || 'Execution failed');
            callback(null, response);
        }
    }
    /**
     * Stop the server
     */
    stop() {
        this.server.forceShutdown();
        console.log('🛑 Executor server stopped');
    }
}
exports.ExecutorServer = ExecutorServer;
