// @backend-ai/sdk-express/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoreClient = exports.SchemaGenerator = exports.RouteDiscovery = exports.BackendAI = void 0;
const discovery_1 = require("./discovery");
const schema_generator_1 = require("./schema-generator");
const core_client_1 = require("./core-client");
const router_map_1 = require("./router-map");
const executor_server_1 = require("./executor-server");
/**
 * Backend AI SDK for Express
 * Auto-discover Express routes and register them as AI tools
 */
class BackendAI {
    /**
     * Initialize the SDK with an Express app
     */
    static async init(app, options) {
        // Validate options
        if (!options.adapterId || !options.tenantId || !options.coreUrl) {
            throw new Error('adapterId, tenantId, and coreUrl are required');
        }
        const sdk = new BackendAI();
        sdk.options = {
            autoDiscover: true,
            persist: true,
            executorPort: 6000,
            geminiModel: 'gemini-2.5-flash',
            verbose: false,
            ...options,
        };
        console.log('🚀 Initializing Backend AI SDK for Express...');
        try {
            // 1. Discover routes
            if (sdk.options.autoDiscover !== false) {
                const discovery = new discovery_1.RouteDiscovery();
                const routes = discovery.discover(app);
                if (sdk.options.verbose) {
                    console.log(`🔍 Discovered ${routes.length} routes:`);
                    routes.forEach((r) => console.log(`   ${r.method} ${r.path}`));
                }
                else {
                    console.log(`🔍 Discovered ${routes.length} Express routes`);
                }
                if (routes.length === 0) {
                    console.warn('⚠️ No routes discovered. Make sure your Express app has routes defined.');
                    return;
                }
                // 2. Generate schemas using AI
                console.log('🤖 Generating tool schemas using Gemini AI...');
                const generator = new schema_generator_1.SchemaGenerator(sdk.options.geminiApiKey, sdk.options.geminiModel, sdk.options.verbose);
                const tools = await generator.generateSchemas(routes);
                console.log(`✅ Generated ${tools.length} tool definitions`);
                // 3. Register with Core (if persist is enabled)
                if (sdk.options.persist !== false) {
                    console.log('📡 Registering tools with Core...');
                    sdk.coreClient = new core_client_1.CoreClient(sdk.options.verbose || false);
                    sdk.coreClient.connect(sdk.options.coreUrl);
                    await sdk.coreClient.registerTools(sdk.options.tenantId, sdk.options.adapterId, tools);
                    console.log(`✅ Registered ${tools.length} tools with Core`);
                }
                // 4. Build router map
                const routerMap = new router_map_1.RouterMap();
                const toolIds = tools.map((t) => t.tool_id);
                routerMap.build(routes, toolIds);
                // 5. Start executor server
                sdk.executorServer = new executor_server_1.ExecutorServer(routerMap, sdk.options.executorPort, sdk.options.verbose || false);
                sdk.executorServer.start();
            }
            BackendAI.instance = sdk;
            console.log('✨ Backend AI SDK initialized successfully!');
        }
        catch (error) {
            console.error('❌ SDK initialization failed:', error);
            throw error;
        }
    }
    /**
     * Get the current SDK instance
     */
    static getInstance() {
        return BackendAI.instance;
    }
    /**
     * Shutdown the SDK
     */
    static async shutdown() {
        if (BackendAI.instance?.executorServer) {
            BackendAI.instance.executorServer.stop();
        }
        BackendAI.instance = null;
        console.log('👋 Backend AI SDK shutdown complete');
    }
}
exports.BackendAI = BackendAI;
BackendAI.instance = null;
// Export types
__exportStar(require("./types"), exports);
var discovery_2 = require("./discovery");
Object.defineProperty(exports, "RouteDiscovery", { enumerable: true, get: function () { return discovery_2.RouteDiscovery; } });
var schema_generator_2 = require("./schema-generator");
Object.defineProperty(exports, "SchemaGenerator", { enumerable: true, get: function () { return schema_generator_2.SchemaGenerator; } });
var core_client_2 = require("./core-client");
Object.defineProperty(exports, "CoreClient", { enumerable: true, get: function () { return core_client_2.CoreClient; } });
