// @backend-ai/sdk-express/dist/router-map.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouterMap = void 0;
/**
 * Router Map
 * Maps tool IDs to Express route handlers for execution
 */
class RouterMap {
    constructor() {
        this.map = new Map();
    }
    /**
     * Build router map from discovered routes
     */
    build(routes, toolIds) {
        routes.forEach((route, index) => {
            const toolId = toolIds[index];
            if (toolId) {
                this.map.set(toolId, {
                    method: route.method,
                    path: route.path,
                    handler: route.handler,
                });
            }
        });
    }
    /**
     * Get route mapping by tool ID
     */
    get(toolId) {
        return this.map.get(toolId);
    }
    /**
     * Get all mapped tool IDs
     */
    getToolIds() {
        return Array.from(this.map.keys());
    }
    /**
     * Clear the map
     */
    clear() {
        this.map.clear();
    }
}
exports.RouterMap = RouterMap;
