// @backend-ai/sdk-express/protos/agent_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var agent_pb = require('./agent_pb.js');
var google_protobuf_struct_pb = require('google-protobuf/google/protobuf/struct_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');

function serialize_agent_AgentRequest(arg) {
  if (!(arg instanceof agent_pb.AgentRequest)) {
    throw new Error('Expected argument of type agent.AgentRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_agent_AgentRequest(buffer_arg) {
  return agent_pb.AgentRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_agent_AgentResponse(arg) {
  if (!(arg instanceof agent_pb.AgentResponse)) {
    throw new Error('Expected argument of type agent.AgentResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_agent_AgentResponse(buffer_arg) {
  return agent_pb.AgentResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_agent_ExecutionEvent(arg) {
  if (!(arg instanceof agent_pb.ExecutionEvent)) {
    throw new Error('Expected argument of type agent.ExecutionEvent');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_agent_ExecutionEvent(buffer_arg) {
  return agent_pb.ExecutionEvent.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_agent_ExecutionStreamRequest(arg) {
  if (!(arg instanceof agent_pb.ExecutionStreamRequest)) {
    throw new Error('Expected argument of type agent.ExecutionStreamRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_agent_ExecutionStreamRequest(buffer_arg) {
  return agent_pb.ExecutionStreamRequest.deserializeBinary(new Uint8Array(buffer_arg));
}


// ===== Agent Service =====
var AgentServiceService = exports.AgentServiceService = {
  // Entry point: user prompt → plan → execution
processPrompt: {
    path: '/agent.AgentService/ProcessPrompt',
    requestStream: false,
    responseStream: false,
    requestType: agent_pb.AgentRequest,
    responseType: agent_pb.AgentResponse,
    requestSerialize: serialize_agent_AgentRequest,
    requestDeserialize: deserialize_agent_AgentRequest,
    responseSerialize: serialize_agent_AgentResponse,
    responseDeserialize: deserialize_agent_AgentResponse,
  },
  // Streaming execution updates
streamExecution: {
    path: '/agent.AgentService/StreamExecution',
    requestStream: false,
    responseStream: true,
    requestType: agent_pb.ExecutionStreamRequest,
    responseType: agent_pb.ExecutionEvent,
    requestSerialize: serialize_agent_ExecutionStreamRequest,
    requestDeserialize: deserialize_agent_ExecutionStreamRequest,
    responseSerialize: serialize_agent_ExecutionEvent,
    responseDeserialize: deserialize_agent_ExecutionEvent,
  },
};

exports.AgentServiceClient = grpc.makeGenericClientConstructor(AgentServiceService, 'AgentService');
