// @backend-ai/sdk-express/protos/auth_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var auth_pb = require('./auth_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');

function serialize_auth_AuthRequest(arg) {
  if (!(arg instanceof auth_pb.AuthRequest)) {
    throw new Error('Expected argument of type auth.AuthRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_auth_AuthRequest(buffer_arg) {
  return auth_pb.AuthRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_auth_AuthResponse(arg) {
  if (!(arg instanceof auth_pb.AuthResponse)) {
    throw new Error('Expected argument of type auth.AuthResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_auth_AuthResponse(buffer_arg) {
  return auth_pb.AuthResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// ===== Auth Service =====
var AuthServiceService = exports.AuthServiceService = {
  // SDK → Core: validate token before execution
validateToken: {
    path: '/auth.AuthService/ValidateToken',
    requestStream: false,
    responseStream: false,
    requestType: auth_pb.AuthRequest,
    responseType: auth_pb.AuthResponse,
    requestSerialize: serialize_auth_AuthRequest,
    requestDeserialize: deserialize_auth_AuthRequest,
    responseSerialize: serialize_auth_AuthResponse,
    responseDeserialize: deserialize_auth_AuthResponse,
  },
};

exports.AuthServiceClient = grpc.makeGenericClientConstructor(AuthServiceService, 'AuthService');
