// @backend-ai/sdk-express/protos/control_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var control_pb = require('./control_pb.js');

function serialize_control_RegisterAdapterRequest(arg) {
  if (!(arg instanceof control_pb.RegisterAdapterRequest)) {
    throw new Error('Expected argument of type control.RegisterAdapterRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_control_RegisterAdapterRequest(buffer_arg) {
  return control_pb.RegisterAdapterRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_control_RegisterAdapterResponse(arg) {
  if (!(arg instanceof control_pb.RegisterAdapterResponse)) {
    throw new Error('Expected argument of type control.RegisterAdapterResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_control_RegisterAdapterResponse(buffer_arg) {
  return control_pb.RegisterAdapterResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_control_SyncToolsRequest(arg) {
  if (!(arg instanceof control_pb.SyncToolsRequest)) {
    throw new Error('Expected argument of type control.SyncToolsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_control_SyncToolsRequest(buffer_arg) {
  return control_pb.SyncToolsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_control_SyncToolsResponse(arg) {
  if (!(arg instanceof control_pb.SyncToolsResponse)) {
    throw new Error('Expected argument of type control.SyncToolsResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_control_SyncToolsResponse(buffer_arg) {
  return control_pb.SyncToolsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var ControlServiceService = exports.ControlServiceService = {
  registerAdapter: {
    path: '/control.ControlService/RegisterAdapter',
    requestStream: false,
    responseStream: false,
    requestType: control_pb.RegisterAdapterRequest,
    responseType: control_pb.RegisterAdapterResponse,
    requestSerialize: serialize_control_RegisterAdapterRequest,
    requestDeserialize: deserialize_control_RegisterAdapterRequest,
    responseSerialize: serialize_control_RegisterAdapterResponse,
    responseDeserialize: deserialize_control_RegisterAdapterResponse,
  },
  syncTools: {
    path: '/control.ControlService/SyncTools',
    requestStream: false,
    responseStream: false,
    requestType: control_pb.SyncToolsRequest,
    responseType: control_pb.SyncToolsResponse,
    requestSerialize: serialize_control_SyncToolsRequest,
    requestDeserialize: deserialize_control_SyncToolsRequest,
    responseSerialize: serialize_control_SyncToolsResponse,
    responseDeserialize: deserialize_control_SyncToolsResponse,
  },
};

exports.ControlServiceClient = grpc.makeGenericClientConstructor(ControlServiceService, 'ControlService');
