// @backend-ai/sdk-express/protos/execute_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var execute_pb = require('./execute_pb.js');
var google_protobuf_struct_pb = require('google-protobuf/google/protobuf/struct_pb.js');

function serialize_execute_ToolExecutionRequest(arg) {
  if (!(arg instanceof execute_pb.ToolExecutionRequest)) {
    throw new Error('Expected argument of type execute.ToolExecutionRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_execute_ToolExecutionRequest(buffer_arg) {
  return execute_pb.ToolExecutionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_execute_ToolExecutionResponse(arg) {
  if (!(arg instanceof execute_pb.ToolExecutionResponse)) {
    throw new Error('Expected argument of type execute.ToolExecutionResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_execute_ToolExecutionResponse(buffer_arg) {
  return execute_pb.ToolExecutionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var ToolExecutionServiceService = exports.ToolExecutionServiceService = {
  executeTool: {
    path: '/execute.ToolExecutionService/ExecuteTool',
    requestStream: false,
    responseStream: false,
    requestType: execute_pb.ToolExecutionRequest,
    responseType: execute_pb.ToolExecutionResponse,
    requestSerialize: serialize_execute_ToolExecutionRequest,
    requestDeserialize: deserialize_execute_ToolExecutionRequest,
    responseSerialize: serialize_execute_ToolExecutionResponse,
    responseDeserialize: deserialize_execute_ToolExecutionResponse,
  },
};

exports.ToolExecutionServiceClient = grpc.makeGenericClientConstructor(ToolExecutionServiceService, 'ToolExecutionService');
