// @backend-ai/sdk-express/protos/streaming_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var streaming_pb = require('./streaming_pb.js');
var google_protobuf_struct_pb = require('google-protobuf/google/protobuf/struct_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');

function serialize_streaming_StreamAck(arg) {
  if (!(arg instanceof streaming_pb.StreamAck)) {
    throw new Error('Expected argument of type streaming.StreamAck');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_streaming_StreamAck(buffer_arg) {
  return streaming_pb.StreamAck.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_streaming_StreamEvent(arg) {
  if (!(arg instanceof streaming_pb.StreamEvent)) {
    throw new Error('Expected argument of type streaming.StreamEvent');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_streaming_StreamEvent(buffer_arg) {
  return streaming_pb.StreamEvent.deserializeBinary(new Uint8Array(buffer_arg));
}


// ===== Streaming Service =====
var StreamingServiceService = exports.StreamingServiceService = {
  // SDK → Core: stream execution events
publishEvents: {
    path: '/streaming.StreamingService/PublishEvents',
    requestStream: true,
    responseStream: false,
    requestType: streaming_pb.StreamEvent,
    responseType: streaming_pb.StreamAck,
    requestSerialize: serialize_streaming_StreamEvent,
    requestDeserialize: deserialize_streaming_StreamEvent,
    responseSerialize: serialize_streaming_StreamAck,
    responseDeserialize: deserialize_streaming_StreamAck,
  },
};

exports.StreamingServiceClient = grpc.makeGenericClientConstructor(StreamingServiceService, 'StreamingService');
