// @backend-ai/sdk-express/protos/tool_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var tool_pb = require('./tool_pb.js');
var google_protobuf_struct_pb = require('google-protobuf/google/protobuf/struct_pb.js');

function serialize_tool_RegisterToolsRequest(arg) {
  if (!(arg instanceof tool_pb.RegisterToolsRequest)) {
    throw new Error('Expected argument of type tool.RegisterToolsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tool_RegisterToolsRequest(buffer_arg) {
  return tool_pb.RegisterToolsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tool_RegisterToolsResponse(arg) {
  if (!(arg instanceof tool_pb.RegisterToolsResponse)) {
    throw new Error('Expected argument of type tool.RegisterToolsResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tool_RegisterToolsResponse(buffer_arg) {
  return tool_pb.RegisterToolsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var ToolRegistryServiceService = exports.ToolRegistryServiceService = {
  registerTools: {
    path: '/tool.ToolRegistryService/RegisterTools',
    requestStream: false,
    responseStream: false,
    requestType: tool_pb.RegisterToolsRequest,
    responseType: tool_pb.RegisterToolsResponse,
    requestSerialize: serialize_tool_RegisterToolsRequest,
    requestDeserialize: deserialize_tool_RegisterToolsRequest,
    responseSerialize: serialize_tool_RegisterToolsResponse,
    responseDeserialize: deserialize_tool_RegisterToolsResponse,
  },
};

exports.ToolRegistryServiceClient = grpc.makeGenericClientConstructor(ToolRegistryServiceService, 'ToolRegistryService');
