// @backend-ai/sdk-express/src/core-client.ts
import * as grpc from '@grpc/grpc-js';
import { ToolDefinition } from './types';

// Load proto files from SDK package directory
const path = require('path');
const protoDir = path.join(__dirname, '../protos');
const toolGrpc = require(path.join(protoDir, 'tool_grpc_pb'));
const toolProto = require(path.join(protoDir, 'tool_pb'));
const Struct = require('google-protobuf/google/protobuf/struct_pb').Struct;

/**
 * Core Service Client
 * Handles communication with the Backend AI Core service
 */
export class CoreClient {
    private client: any;
    private verbose: boolean;

    constructor(verbose: boolean = false) {
        this.verbose = verbose;
    }

    /**
     * Connect to Core service
     */
    connect(coreUrl: string): void {
        try {
            this.client = new toolGrpc.ToolRegistryServiceClient(
                coreUrl,
                grpc.credentials.createInsecure()
            );

            if (this.verbose) {
                console.log(`✅ Connected to Core at ${coreUrl}`);
            }
        } catch (error) {
            console.error('❌ Failed to connect to Core:', error);
            throw error;
        }
    }

    /**
     * Register tools with Core
     */
    async registerTools(
        tenantId: string,
        adapterId: string,
        tools: ToolDefinition[]
    ): Promise<void> {
        return new Promise((resolve, reject) => {
            try {
                const req = new toolProto.RegisterToolsRequest();
                req.setTenantId(tenantId);
                req.setAdapterId(adapterId);

                const toolDefs = tools.map((t) => {
                    const def = new toolProto.ToolDefinition();

                    def.setToolId(t.tool_id);
                    def.setName(t.name);
                    def.setDescription(t.description);

                    // Convert schemas to Struct
                    const inputStruct = Struct.fromJavaScript(t.input_schema || {});
                    const outputStruct = Struct.fromJavaScript(t.response_schema || {});

                    def.setInputSchema(inputStruct);
                    def.setOutputSchema(outputStruct);

                    return def;
                });

                req.setToolsList(toolDefs);

                this.client.registerTools(req, (err: any, response: any) => {
                    if (err) {
                        console.error('❌ Tool registration failed:', err);
                        return reject(err);
                    }

                    if (this.verbose) {
                        console.log(`✅ Successfully registered ${tools.length} tools with Core`);
                    }

                    resolve();
                });
            } catch (error) {
                console.error('❌ Error during tool registration:', error);
                reject(error);
            }
        });
    }

    /**
     * Verify connection to Core
     */
    async ping(): Promise<boolean> {
        try {
            // Implement ping if Core has health check endpoint
            return true;
        } catch {
            return false;
        }
    }
}
