// @backend-ai/sdk-express/src/discovery.ts
import { Application } from 'express';
import { DiscoveredRoute } from './types';

/**
 * Route Discovery Service
 * Automatically discovers all Express routes from an Express app instance
 */
export class RouteDiscovery {
    /**
     * Discover all routes from Express app
     */
    discover(app: Application): DiscoveredRoute[] {
        const routes: DiscoveredRoute[] = [];

        // Access Express internal router stack
        const stack = (app as any)._router?.stack || [];

        this.walkStack(stack, '', routes);

        return routes;
    }

    /**
     * Recursively walk through Express router stack
     */
    private walkStack(
        stack: any[],
        prefix: string,
        routes: DiscoveredRoute[]
    ): void {
        stack.forEach((layer: any) => {
            if (layer.route) {
                // Direct route handler
                const path = prefix + layer.route.path;
                const methods = layer.route.methods;

                Object.keys(methods).forEach((method) => {
                    if (methods[method]) {
                        const handler = layer.route.stack[0]?.handle;

                        if (handler) {
                            routes.push({
                                method: method.toUpperCase(),
                                path: this.normalizePath(path),
                                handler: handler,
                                handlerName: handler.name || 'anonymous',
                            });
                        }
                    }
                });
            } else if (layer.name === 'router' && layer.handle?.stack) {
                // Nested router
                let routerPrefix = this.extractRouterPrefix(layer);
                this.walkStack(layer.handle.stack, prefix + routerPrefix, routes);
            }
        });
    }

    /**
     * Extract prefix from nested router
     */
    private extractRouterPrefix(layer: any): string {
        if (!layer.regexp) return '';

        // Try to extract path from regexp
        const regexpSource = layer.regexp.source;

        // Match patterns like: ^\\/products\\/
        const match = regexpSource.match(/^\^\\\\\/([^\\\/\?]+)/);

        if (match) {
            return '/' + match[1];
        }

        return '';
    }

    /**
     * Normalize path (remove duplicate slashes)
     */
    private normalizePath(path: string): string {
        return path.replace(/\/+/g, '/');
    }
}
