// @backend-ai/sdk-express/src/executor-server.ts
import * as grpc from '@grpc/grpc-js';
import { RouterMap } from './router-map';
import { ExpressExecutor } from './executor';

// Load proto files from SDK package directory
const path = require('path');
const protoDir = path.join(__dirname, '../protos');
const executeGrpc = require(path.join(protoDir, 'execute_grpc_pb'));
const executeProto = require(path.join(protoDir, 'execute_pb'));
const Struct = require('google-protobuf/google/protobuf/struct_pb').Struct;

/**
 * gRPC Executor Server
 * Receives tool execution requests from Core and executes Express handlers
 */
export class ExecutorServer {
    private server: grpc.Server;
    private executor: ExpressExecutor;
    private verbose: boolean;

    constructor(
        routerMap: RouterMap,
        private port: number = 6000,
        verbose: boolean = false
    ) {
        this.server = new grpc.Server();
        this.executor = new ExpressExecutor(routerMap);
        this.verbose = verbose;
    }

    /**
     * Start the gRPC server
     */
    start(): void {
        // Register service
        this.server.addService(executeGrpc.ToolExecutionServiceService, {
            executeTool: this.handleExecuteTool.bind(this),
        });

        // Bind to port
        this.server.bindAsync(
            `0.0.0.0:${this.port}`,
            grpc.ServerCredentials.createInsecure(),
            (err, port) => {
                if (err) {
                    console.error('❌ Failed to start executor server:', err);
                    throw err;
                }

                console.log(`🧩 SDK Executor running on port ${port}`);
            }
        );
    }

    /**
     * Handle tool execution request
     */
    private async handleExecuteTool(call: any, callback: any): Promise<void> {
        try {
            const toolId = call.request.getToolId();
            const inputStruct = call.request.getInput();

            const input = inputStruct ? inputStruct.toJavaScript() : {};

            if (this.verbose) {
                console.log(`[Executor] Executing tool: ${toolId}`);
                console.log(`[Executor] Input:`, input);
            }

            // Execute the tool
            const result = await this.executor.execute(toolId, input);

            if (this.verbose) {
                console.log(`[Executor] Result:`, result);
            }

            // Create response
            const outputStruct = Struct.fromJavaScript(result || {});
            const response = new executeProto.ToolExecutionResponse();
            response.setSuccess(true);
            response.setOutput(outputStruct);

            callback(null, response);
        } catch (err: any) {
            console.error('❌ Tool execution failed:', err);

            const response = new executeProto.ToolExecutionResponse();
            response.setSuccess(false);
            response.setErrorMessage(err?.message || 'Execution failed');

            callback(null, response);
        }
    }

    /**
     * Stop the server
     */
    stop(): void {
        this.server.forceShutdown();
        console.log('🛑 Executor server stopped');
    }
}
