// @backend-ai/sdk-express/src/executor.ts
import { RouterMap } from './router-map';

/**
 * Express Executor
 * Executes Express route handlers with mock req/res objects
 */
export class ExpressExecutor {
    constructor(private routerMap: RouterMap) { }

    /**
     * Execute a tool by ID
     */
    async execute(toolId: string, input: any): Promise<any> {
        const mapping = this.routerMap.get(toolId);

        if (!mapping) {
            throw new Error(`Tool not found: ${toolId}`);
        }

        const { handler, path, method } = mapping;

        // Parse path parameters
        const paramKeys = this.extractParamKeys(path);

        // Distribute input to req parts
        const { params, query, body } = this.distributeInput(
            input,
            paramKeys,
            method
        );

        // Create mock Express request
        const req: any = {
            body,
            params,
            query,
            headers: {},
            method,
            path,
        };

        // Create mock Express response
        return new Promise((resolve, reject) => {
            let status = 200;

            const res: any = {
                status(code: number) {
                    status = code;
                    return this;
                },

                json: (data: any) => {
                    const safeData =
                        data && typeof data === 'object'
                            ? JSON.parse(JSON.stringify(data))
                            : {};

                    resolve({
                        status,
                        body: safeData,
                    });
                },

                send: (data: any) => {
                    const safeData =
                        data && typeof data === 'object'
                            ? JSON.parse(JSON.stringify(data))
                            : {};

                    resolve({
                        status,
                        body: safeData,
                    });
                },
            };

            try {
                const result = handler(req, res);

                // Handle async handlers
                if (result instanceof Promise) {
                    result.catch(reject);
                }
            } catch (err) {
                reject(err);
            }
        });
    }

    /**
     * Extract parameter keys from path
     */
    private extractParamKeys(path: string): string[] {
        const keys: string[] = [];
        const parts = path.split('/');

        for (const part of parts) {
            if (part.startsWith(':')) {
                const key = part.substring(1).replace('?', '');
                keys.push(key);
            }
        }

        return keys;
    }

    /**
     * Distribute input to req.params, req.query, req.body
     */
    private distributeInput(
        input: any,
        paramKeys: string[],
        method: string
    ): { params: any; query: any; body: any } {
        const params: any = {};
        const query: any = {};
        const body: any = {};

        const rawInput = input?.body || input || {};

        for (const [key, value] of Object.entries(rawInput)) {
            if (paramKeys.includes(key)) {
                params[key] = value;
            } else if (method === 'GET' || method === 'DELETE') {
                query[key] = value;
            } else {
                body[key] = value;
            }
        }

        return { params, query, body };
    }
}
