// @backend-ai/sdk-express/src/index.ts
import { Application } from 'express';
import { BackendAIOptions, ToolDefinition } from './types';
import { RouteDiscovery } from './discovery';
import { SchemaGenerator } from './schema-generator';
import { CoreClient } from './core-client';
import { RouterMap } from './router-map';
import { ExecutorServer } from './executor-server';

/**
 * Backend AI SDK for Express
 * Auto-discover Express routes and register them as AI tools
 */
export class BackendAI {
    private static instance: BackendAI | null = null;
    private executorServer?: ExecutorServer;
    private coreClient?: CoreClient;
    private options?: BackendAIOptions;

    /**
     * Initialize the SDK with an Express app
     */
    static async init(
        app: Application,
        options: BackendAIOptions
    ): Promise<void> {
        // Validate options
        if (!options.adapterId || !options.tenantId || !options.coreUrl) {
            throw new Error('adapterId, tenantId, and coreUrl are required');
        }

        const sdk = new BackendAI();
        sdk.options = {
            autoDiscover: true,
            persist: true,
            executorPort: 6000,
            geminiModel: 'gemini-2.5-flash',
            verbose: false,
            ...options,
        };

        console.log('🚀 Initializing Backend AI SDK for Express...');

        try {
            // 1. Discover routes
            if (sdk.options.autoDiscover !== false) {
                const discovery = new RouteDiscovery();
                const routes = discovery.discover(app);

                if (sdk.options.verbose) {
                    console.log(`🔍 Discovered ${routes.length} routes:`);
                    routes.forEach((r) => console.log(`   ${r.method} ${r.path}`));
                } else {
                    console.log(`🔍 Discovered ${routes.length} Express routes`);
                }

                if (routes.length === 0) {
                    console.warn('⚠️ No routes discovered. Make sure your Express app has routes defined.');
                    return;
                }

                // 2. Generate schemas using AI
                console.log('🤖 Generating tool schemas using Gemini AI...');
                const generator = new SchemaGenerator(
                    sdk.options.geminiApiKey,
                    sdk.options.geminiModel,
                    sdk.options.verbose
                );

                const tools = await generator.generateSchemas(routes);
                console.log(`✅ Generated ${tools.length} tool definitions`);

                // 3. Register with Core (if persist is enabled)
                if (sdk.options.persist !== false) {
                    console.log('📡 Registering tools with Core...');
                    sdk.coreClient = new CoreClient(sdk.options.verbose || false);
                    sdk.coreClient.connect(sdk.options.coreUrl);

                    await sdk.coreClient.registerTools(
                        sdk.options.tenantId,
                        sdk.options.adapterId,
                        tools
                    );

                    console.log(`✅ Registered ${tools.length} tools with Core`);
                }

                // 4. Build router map
                const routerMap = new RouterMap();
                const toolIds = tools.map((t) => t.tool_id);
                routerMap.build(routes, toolIds);

                // 5. Start executor server
                sdk.executorServer = new ExecutorServer(
                    routerMap,
                    sdk.options.executorPort,
                    sdk.options.verbose || false
                );

                sdk.executorServer.start();
            }

            BackendAI.instance = sdk;
            console.log('✨ Backend AI SDK initialized successfully!');
        } catch (error) {
            console.error('❌ SDK initialization failed:', error);
            throw error;
        }
    }

    /**
     * Get the current SDK instance
     */
    static getInstance(): BackendAI | null {
        return BackendAI.instance;
    }

    /**
     * Shutdown the SDK
     */
    static async shutdown(): Promise<void> {
        if (BackendAI.instance?.executorServer) {
            BackendAI.instance.executorServer.stop();
        }
        BackendAI.instance = null;
        console.log('👋 Backend AI SDK shutdown complete');
    }
}

// Export types
export * from './types';
export { RouteDiscovery } from './discovery';
export { SchemaGenerator } from './schema-generator';
export { CoreClient } from './core-client';
