// @backend-ai/sdk-express/src/router-map.ts
import { DiscoveredRoute, RouteMapping } from './types';

/**
 * Router Map
 * Maps tool IDs to Express route handlers for execution
 */
export class RouterMap {
    private map: Map<string, RouteMapping> = new Map();

    /**
     * Build router map from discovered routes
     */
    build(routes: DiscoveredRoute[], toolIds: string[]): void {
        routes.forEach((route, index) => {
            const toolId = toolIds[index];

            if (toolId) {
                this.map.set(toolId, {
                    method: route.method,
                    path: route.path,
                    handler: route.handler,
                });
            }
        });
    }

    /**
     * Get route mapping by tool ID
     */
    get(toolId: string): RouteMapping | undefined {
        return this.map.get(toolId);
    }

    /**
     * Get all mapped tool IDs
     */
    getToolIds(): string[] {
        return Array.from(this.map.keys());
    }

    /**
     * Clear the map
     */
    clear(): void {
        this.map.clear();
    }
}
