// @backend-ai/sdk-express/src/types.ts
import { Application } from 'express';

/**
 * SDK Initialization Options
 */
export interface BackendAIOptions {
    /** Unique identifier for this adapter instance */
    adapterId: string;

    /** Tenant ID for multi-tenancy support */
    tenantId: string;

    /** Core service URL (e.g., 'localhost:50051') */
    coreUrl: string;

    /** Port for the gRPC executor server (default: 6000) */
    executorPort?: number;

    /** Auto-discover routes on initialization (default: true) */
    autoDiscover?: boolean;

    /** Persist schemas to Core database (default: true) */
    persist?: boolean;

    /** Gemini API key for schema generation (optional, falls back to env var) */
    geminiApiKey?: string;

    /** Gemini model to use (default: 'gemini-2.5-flash') */
    geminiModel?: string;

    /** Enable verbose logging (default: false) */
    verbose?: boolean;
}

/**
 * Discovered Express Route
 */
export interface DiscoveredRoute {
    /** HTTP method (GET, POST, PUT, DELETE, etc.) */
    method: string;

    /** Route path (e.g., '/products/:id') */
    path: string;

    /** Route handler function */
    handler: Function;

    /** Handler function name */
    handlerName: string;
}

/**
 * Tool Definition sent to Core
 */
export interface ToolDefinition {
    /** Unique tool identifier */
    tool_id: string;

    /** Human-readable tool name */
    name: string;

    /** Tool description */
    description: string;

    /** Input parameter schema */
    input_schema: {
        type: 'object';
        description: string;
        properties: Record<string, any>;
        required: string[];
    };

    /** Response schema */
    response_schema: {
        type: 'object';
        description: string;
        properties: Record<string, any>;
    };

    /** Adapter metadata */
    adapter: {
        framework: 'express';
        method: string;
        path: string;
    };
}

/**
 * Route mapping for execution
 */
export interface RouteMapping {
    method: string;
    path: string;
    handler: Function;
}
