// @backend-ai/sdk-express/dist/protos/adapter_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var adapter_pb = require('./adapter_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');

function serialize_adapter_RegisterAdapterRequest(arg) {
  if (!(arg instanceof adapter_pb.RegisterAdapterRequest)) {
    throw new Error('Expected argument of type adapter.RegisterAdapterRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_adapter_RegisterAdapterRequest(buffer_arg) {
  return adapter_pb.RegisterAdapterRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_adapter_RegisterAdapterResponse(arg) {
  if (!(arg instanceof adapter_pb.RegisterAdapterResponse)) {
    throw new Error('Expected argument of type adapter.RegisterAdapterResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_adapter_RegisterAdapterResponse(buffer_arg) {
  return adapter_pb.RegisterAdapterResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var AdapterRegistryServiceService = exports.AdapterRegistryServiceService = {
  registerAdapter: {
    path: '/adapter.AdapterRegistryService/RegisterAdapter',
    requestStream: false,
    responseStream: false,
    requestType: adapter_pb.RegisterAdapterRequest,
    responseType: adapter_pb.RegisterAdapterResponse,
    requestSerialize: serialize_adapter_RegisterAdapterRequest,
    requestDeserialize: deserialize_adapter_RegisterAdapterRequest,
    responseSerialize: serialize_adapter_RegisterAdapterResponse,
    responseDeserialize: deserialize_adapter_RegisterAdapterResponse,
  },
};

exports.AdapterRegistryServiceClient = grpc.makeGenericClientConstructor(AdapterRegistryServiceService, 'AdapterRegistryService');
